### PR TITLE
Github actions

### PR DESCRIPTION
**Pull request by scotthavens**
Migrate from Travis CI to Github Actions. This will also deploy to Pypi when a release is made.

### CODE REVIEW

**Comment by scotthavens** [comment]
Tested with a release `v0.7.0-rc` which built and was published to Pypi.

https://pypi.org/project/weather-forecast-retrieval/0.7.0rc0/

File: tests/test_hrrr_preprocessor.py [hunk 1]
```diff
@@ -62,6 +68,6 @@ def test_pre_process(self):
         )
 
         self.assertCountEqual(
-            data.keys(),
+            list(data.keys()),
```
Comment (jomey) [inline tests/test_hrrr_preprocessor.py:71]: I just changes this and removed the cast to a list. Doesn't the `keys()` method already return one?

Comment (scotthavens) [reply tests/test_hrrr_preprocessor.py:71]: Maybe in newer python versions. Python 3.6 returns a `dict_keys` list which is not equal when using `assertListEqual`. But comparing with `assertCountEqual` does work.

Comment (jomey) [reply tests/test_hrrr_preprocessor.py]: Think there is not really a need to compare the order and mainly make sure that the keys are present. Hence my earlier change to `CountEquals`

**Review by jomey — APPROVED** [review]

File: tests/test_hrrr_preprocessor.py (hunk 1 above)
Comment (scotthavens) [reply tests/test_hrrr_preprocessor.py]: Must have been a bad merge on my part then! Looks like we're back to normal.

**Comment by jomey** [comment]
Really like the new logic and integration with GH actions

**Pull request merged**